### PR TITLE
Track unmappable dims correctly in compute at.

### LINKED ISSUE #1305: fusion expression sorting failure
## 🐛 Bug

Reported by functorch/AOTAutograd folks where they are manually (?! not totally sure) fusing fwd & bwd ops and gives us an example where codegen fails.

Error message:
```
root@d1eed84a4cff:/raid/playground# PYTORCH_NVFUSER_DISABLE_FALLBACK=1 PYTORCH_JIT_LOG_LEVEL=kernel_cache python horace_repro_simplified.py
[DUMP kernel_cache.cpp:630] GraphCache::runGraphWithInputs graph:
[DUMP kernel_cache.cpp:630] graph(%0 : Float(128, strides=[1], requires_grad=0, device=cuda:0),
[DUMP kernel_cache.cpp:630]       %1 : Float(128, strides=[1], requires_grad=0, device=cuda:0),
[DUMP kernel_cache.cpp:630]       %2 : Float(8192, 128, strides=[128, 1], requires_grad=0, device=cuda:0),
[DUMP kernel_cache.cpp:630]       %3 : Float(128, strides=[1], requires_grad=0, device=cuda:0),
[DUMP kernel_cache.cpp:630]       %4 : Float(8192, 128, strides=[128, 1], requires_grad=0, device=cuda:0)):
[DUMP kernel_cache.cpp:630]   %5 : bool[] = prim::Constant[value=[True, True, True]]()
[DUMP kernel_cache.cpp:630]   %6 : float = prim::Constant[value=1.]() # horace_repro_simplified.py:45:19
[DUMP kernel_cache.cpp:630]   %7 : float = prim::Constant[value=1.0000000000000001e-05]() # horace_repro_simplified.py:41:76
[DUMP kernel_cache.cpp:630]   %8 : int[] = prim::Constant[value=[128]]()
[DUMP kernel_cache.cpp:630]   %9 : int = prim::Constant[value=1]()
[DUMP kernel_cache.cpp:630]   %t9.1 : Float(8192, 128, device=cuda:0) = aten::add(%3, %4, %9) # horace_repro_simplified.py:40:9
[DUMP kernel_cache.cpp:630]   %t10.1 : FloatTensor(device=cuda:0), %t11.1 : FloatTensor(device=cuda:0), %t12.1 : FloatTensor(device=cuda:0) = aten::native_layer_norm(%t9.1, %8, %0, %1, %7) # horace_repro_simplified.py:41:20
[DUMP kernel_cache.cpp:630]   %t15.1 : FloatTensor(device=cuda:0) = aten::sigmoid(%t10.1) # horace_repro_simplified.py:42:10
[DUMP kernel_cache.cpp:630]   %t26.1 : Float(8192, 128, device=cuda:0) = aten::mul(%2, %t9.1) # horace_repro_simplified.py:43:10
[DUMP kernel_cache.cpp:630]   %t27.1 : FloatTensor(device=cuda:0) = aten::mul(%2, %t15.1) # horace_repro_simplified.py:44:10
[DUMP kernel_cache.cpp:630]   %t_sub5.1 : FloatTensor(device=cuda:0) = aten::sub(%t15.1, %6, %9) # horace_repro_simplified.py:45:13
[DUMP kernel_cache.cpp:630]   %t_neg5.1 : FloatTensor(device=cuda:0) = aten::neg(%t_sub5.1) # horace_repro_simplified.py:46:13
[DUMP kernel_cache.cpp:630]   %t_mul28.1 : FloatTensor(device=cuda:0) = aten::mul(%t15.1, %t_neg5.1) # horace_repro_simplified.py:47:14
[DUMP kernel_cache.cpp:630]   %t_mul29.1 : FloatTensor(device=cuda:0) = aten::mul(%t26.1, %t_mul28.1) # horace_repro_simplified.py:48:14
[DUMP kernel_cache.cpp:630]   %t20.1 : FloatTensor(device=cuda:0), %t21 : Float(128, strides=[1], requires_grad=0, device=cuda:0), %t22 : Float(128, strides=[1], requires_grad=0, device=cuda:0) = aten::native_layer_norm_backward(%t_mul29.1, %t9.1, %8, %t11.1, %t12.1, %0, %1, %5) # horace_repro_simplified.py:49:20
[DUMP kernel_cache.cpp:630]   %t_add12.1 : FloatTensor(device=cuda:0) = aten::add(%t27.1, %t20.1, %9) # horace_repro_simplified.py:50:14
[DUMP kernel_cache.cpp:630]   return (%t_add12.1)
Traceback (most recent call last):
  File "horace_repro_simplified.py", line 67, in <module>
    out = jit_model(*inps)
RuntimeError: The following operation failed in the TorchScript interpreter.
Traceback of TorchScript (most recent call last):
RuntimeError: !fallback_mode_enabled_INTERNAL ASSERT FAILED at "/raid/pytorch_upstream/torch/csrc/jit/codegen/cuda/lower_expr_sort.cpp":974, please report a bug to PyTorch. Couldn't succcessfully sort out the fusion expressions. There are remaining connections of the heirarchical segmentation which should have been flattened to a single ordered group, or disjoint ordered groups.
```

## To Reproduce

```
import torch                                                                                                               
from torch.nn import Module                                                                                                
import torch.nn.functional as F                                                                                            
                                                                                                                           
torch._C._jit_set_nvfuser_enabled(True)                                                                                    
torch._C._jit_set_texpr_fuser_enabled(False)                                                                               
torch._C._jit_set_profiling_executor(True)                                                                                 
torch._C._jit_set_profiling_mode(True)                                                                                     
torch._C._jit_override_can_fuse_on_cpu(False)                                                                              
torch._C._jit_override_can_fuse_on_gpu(False)                                                                              
torch._C._jit_set_bailout_depth(20)                                                                                        
torch._C._debug_set_autodiff_subgraph_inlining(False)                                                                      
                                                                                                                           
def t(inp0, inp1, inp2, inp3, inp4):                                                                                       
    t9 = inp3 + inp4                                                                                                       
    t10, t11, t12 = torch.ops.aten.native_layer_norm(t9, [128], inp0, inp1, 1e-5)                                          
    t15 = torch.sigmoid(t10)                                                                                               
    t26 = inp2 * t9                                                                                                        
    t27 = inp2 * t15                                                                                                       
    t_sub5 = t15 - 1.0                                                                                                     
    t_neg5 = t_sub5.neg()                                                                                                  
    t_mul28 = t15 * t_neg5                                                                                                 
    t_mul29 = t26 * t_mul28                                                                                                
    t20, t21, t22 = torch.ops.aten.native_layer_norm_backward(t_mul29, t9, [128], t11, t12, inp0, inp1, [True, True, True])
    t_add12 = t27 + t20                                                                                                    
    return t_add12                                                                                                         
                                                                                                                           
if __name__ == "__main__" :                                                                                                
    inp0 = torch.randn(128, device="cuda", dtype=torch.float32)                                                            
    inp1 = torch.randn(128, device="cuda", dtype=torch.float32)                                                            
    inp2 = torch.randn(8192, 128, device="cuda", dtype=torch.float32)                                                      
    inp3 = torch.randn(128, device="cuda", dtype=torch.float32)                                                            
    inp4 = torch.randn(8192, 128, device="cuda", dtype=torch.float32)                                                      
    inps = [inp0, inp1, inp2, inp3, inp4]                                                                                  
                                                                                                                           
    jit_model = torch.jit.script(t)                                                                                                                                                                                
                                                                                                                           
    for idx in range(5) :                                                                                                  
        out = jit_model(*inps)                                                                                             
                                                                                                                           
    print(jit_model.graph_for(*inps)) 
```

### PR DESCRIPTION
**Pull request by csarofeen**
Fixes #1305

See comments in issue for more details.

### CODE REVIEW

**Review by naoyam — APPROVED** [review]
The fix makes sense, but looks like the real problem exists in `ComputeAtRootDomainMap`. I don't understand why `t3` and `t4` are completely inlined. I'll look into it, but feel free to merge this PR if this is urgent.

File: test/cpp/jit/test_gpu.cpp [hunk 1]
```diff
@@ -19604,6 +19604,31 @@ TEST(NVFuserTest, FusionNonDivisibleSplitVectorize2_CUDA) {
   testValidate(&fusion, cg_outputs, {t0}, {ref}, __LINE__, __FILE__);
 }
 
+TEST(NVFuserTest, Issue1305Repro_CUDA) {
```
Comment (naoyam) [inline test/cpp/jit/test_gpu.cpp]: nit: `TEST_F(NVFuserTest, FusionIssue1305Repro_CUDA) {`.

I wonder we really need the `Fusion` suffix.

**Comment by naoyam** [comment]
In the repro, the edge from `T2` to `T4` makes `ComputeAtRootMapBuilder` incorrectly map a pair of `T2` and `T4` and another pair of `T2` and `T3`, resulting in mapping between `T4` and `T3`. This should be fixable.

**Comment by naoyam** [comment]
#1327 fixes the problem with `ComputeAtRootMapBuilder`. I confirmed the repro works fine without the fix in this PR.

**Comment by csarofeen** [comment]
At a minimum this should improve performance. There's bad performance scaling with the number of expressions which does seem to be the compute at pass. So I think this PR can't hurt.

File: test/cpp/jit/test_gpu.cpp (hunk 1 above)
Comment (csarofeen) [reply test/cpp/jit/test_gpu.cpp]: Probably not but best to be consistent

**Pull request merged**